docs/4.4/api-reference.md

```diff
@@ -11,57 +11,328 @@ Teleport is currently working on documenting our API.
+    For production deployments, the `--ttl` flag can be used to ensure a more practical
+    certificate lifetime. See our [Kubernetes Section](kubernetes-ssh.md#using-teleport-kubernetes-with-automation) for more information on automating the signing process for short lived certificates.
+
+## Go Client
+
+Add `client.go` into `/api-examples`.
+
+**client.go**
+
+```go
+{!examples/go-client/client.go!}
+```
+
+## Go Packages
+
+Copy the Teleport module's go.mod below into `/api-examples` and then run `go mod tidy` to slim it down to only what's needed for these api examples.
+
+```
+{!go.mod!}
+```
+
+## Main file
+
+Add this main file to your `/api-examples` folder. Now you can simply plug in the examples below and then run `go run .` to see them in action.
+
+**main.go**
+
+```go
+package main
+
+import (
+	"fmt"
+	"log"
+)
+
+func main() {
+	log.Printf("Starting teleport client...")
+	client, err := connectClient()
+	if err != nil {
+		log.Fatalf("Failed to create client: %v", err)
+	}
+}
+```
+
+## Roles
+
+[Roles](enterprise/ssh-rbac.md#roles) are used to define what resources and actions a user is allowed/denied access to. It's best to make your role's permissions as stringent as possible.
+
+**Retrieve role**
+
+```go
+role, err := client.GetRole("auditor")
+if err != nil {
+  return err
+}
+```
+
+**Create a new role**
+
+```go
+// create a new auditor role which has very limited permissions
+role, err := services.NewRole("auditor", services.RoleSpecV3{
+  Options: services.RoleOptions{
+    MaxSessionTTL: services.Duration(time.Hour),
+  },
+  Allow: services.RoleConditions{
+    Logins: []string{"auditor"},
+    Rules: []services.Rule{
+      // - resources: ['session']
+      //   verbs: ['list', 'read']
+      services.NewRule(services.KindSession, services.RO()),
+    },
+  },
+  Deny: services.RoleConditions{
+    // node_labels: '*': '*'
+    NodeLabels: services.Labels{"*": []string{"*"}},
+  },
+})
+if err != nil {
+  return err
+}
+
+if err = client.UpsertRole(ctx, role); err != nil {
+  return err
+}
+```
+
+**Update role**
+
+```go
+// update the auditor role's ttl to one day
+role.SetOptions(services.RoleOptions{
+	MaxSessionTTL: services.Duration(time.Hour * 24),
+})
+if err := client.UpsertRole(ctx, role); err != nil {
+  return err
+}
+```
 
-## Audit Events API
-Coming Soon
+**Delete role**
+
+```go
+if err := client.DeleteRole(ctx, "auditor"); err != nil {
+  return err
+}
+```
+
+## Tokens
+
+[Tokens](admin-guide.md#adding-nodes-to-the-cluster) are used to add nodes to clusters, and add leaf clusters to [trusted clusters](admin-guide.md#trusted-clusters).
+
+**Retrieve token**
+
+```go
+token, err := client.GetToken(tokenString)
+if err != nil {
+  return err
+}
+```
+
+**Create token**
+
+```go
+tokenString, err := client.GenerateToken(ctx, auth.GenerateTokenRequest{
+  // You can set token explicitly, otherwise it will be generated
+  // Token: CryptoRandomHex()
```

Comment:
If you want to use `CryptoRandomHex()` in this example, include the package it's in and the error handling otherwise it's hard to know what to do with it.

Reply:
This evolved from several pr reviews. Originally it was just a comment saying  "//You can set token explicitly, otherwise it will be generated", then  "// Token: "yourTokenString" // You can set token explicitly, otherwise it will be generated" then it's current iteration. Seems overly complicated and I'm just going to remove the line comments altogether for now.